src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1129,64 +1052,43 @@ static void volume_update_current_vol_ipc4(struct vol_data *cd)
  * \param[in,out] dev Volume base component device.
  * \return Error code.
  */
-static int volume_copy(struct comp_dev *dev)
+static int volume_process(struct processing_module *mod,
+			  struct input_stream_buffer *input_buffers, int num_input_buffers,
+			  struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
-	struct comp_copy_limits c;
-	struct vol_data *cd = comp_get_drvdata(dev);
-	struct comp_buffer *source;
-	struct comp_buffer *sink;
-	uint32_t source_bytes;
-	uint32_t sink_bytes;
+	struct vol_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 	uint32_t frames;
+	uint32_t c_frames = input_buffers[0].size;
```

Comment:
C. was due to the copy limits struct name. Maybe some better descriptive name for this frames variable. Input_frames?